falcon/routing/static.py

```diff
@@ -70,6 +70,21 @@ def _open_range(
     return _BoundedFile(fh, length), length, (start, end, size)
 
 
+def _generate_etag(file_path: str) -> falcon.ETag:
+    """Generate an Etag for a file under a given path.
+
+    Args:
+        file_path (str): Path to the file the ETag should be generated for.
+
+    Returns:
+        falcon.ETag: ETag generated for the file using its modification time and size.
+    """
+    fh = io.open(file_path, 'rb')
+    content_length = os.fstat(fh.fileno()).st_size
```

Comment:
We don't want to run `os.fstat()` multiple times on the same file handle as it is a real syscall.
We only want to call it once per serving, and save the result in a variable.